### PR TITLE
Move runtime_strings.js code into library_strings.js

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -23,6 +23,19 @@ See docs/process.md for more on how version tagging works.
...
+- The following JavaScript runtime functions were converted to JavaScript
+  library functions:
+   - UTF8ArrayToString
+   - UTF8ToString
+   - stringToUTF8Array
+   - stringToUTF8
+   - lengthBytesUTF8
+  If you use any of these functions in your JS code you will now need to include
+  them explictly in one of the following ways:
+   - Add them to a `__deps` entry your JS library file ((with leading $)
```
Comment (kripken) [inline ChangeLog.md:35]:
```suggestion
   - Add them to a `__deps` entry in your JS library file (with leading $)
```

File: src/jsifier.js [hunk 2]
```diff
@@ -203,6 +203,7 @@ function ${name}(${args}) {
       'getWasmTableEntry',
       'runtimeKeepalivePush',
       'runtimeKeepalivePop',
+      'UTF8ToString',
```
Comment (kripken) [inline src/jsifier.js:206]: Why is this here? The comment on 199 says these are deps due to `makeDynCall` or `runtimeKeepalive*` but I don't think either of those generate strings?

File: src/library_legacy.js [hunk 3]
```diff
@@ -44,6 +44,7 @@ mergeInto(LibraryManager.library, {
   // in a maximum length that can be used to be secure from out of bounds
   // writes.
   $writeStringToMemory__docs: '/** @deprecated @param {boolean=} dontAddNull */',
+  $writeStringToMemory__deps: ['$lengthBytesUTF8', '$stringToUTF8Array'],
```
Comment (kripken) [inline src/library_legacy.js:47]:
```suggestion
  $writeStringToMemory__deps: ['$lengthBytesUTF8', '$stringToUTF8'],
```

Comment (kripken) [reply src/library_legacy.js:47]: The comment on line 53 looks out of date, as the code on line 59 uses the non-array version.

File: src/jsifier.js (hunk 2 above)
Comment (sbc100) [reply src/jsifier.js:206]: Sorry, I forget I added this.  Basically there are so many library functions that depend on this that manually adding it to `__deps` seems like more trouble that it was worth.

This is shortcut that automatically adds deps.  Hopefully we can do this for all functions one day.  See #19098.   For now I'm trying to keep this list small and focused.  I'll update the comment.

**Review by kripken — APPROVED** [review]
